web/packages/teleterm/src/ui/Vnet/vnetContext.test.tsx

```diff
@@ -193,15 +211,289 @@ it('registers a callback for unexpected shutdown', async () => {
+          { interval }
+        );
+
+        expect(notificationsService.notifyWarning).toHaveBeenCalledTimes(2);
+        expect(notificationsService.getNotifications()).toHaveLength(1);
+      },
+    },
+    {
+      it: 'is not shown after opening a report, receiving another report with no issues and then issues reoccuring',
+      mockAppContext: appContext => {
+        jest
+          .spyOn(appContext.vnet, 'runDiagnostics')
+          .mockResolvedValueOnce(
+            new MockedUnaryCall({ report: issuesFoundReport })
+          )
+          .mockResolvedValueOnce(
+            new MockedUnaryCall({ report: noIssuesFoundReport })
+          )
+          .mockResolvedValueOnce(
+            new MockedUnaryCall({ report: issuesFoundReport })
+          )
+          .mockResolvedValue(
+            new MockedUnaryCall({}, new Error('something went wrong'))
+          );
+      },
+      verify: async ({ notificationsService, vnet }, result) => {
+        // Open the diag report and verify that it removes the notification.
+        await waitFor(
+          () =>
+            expect(result.current.diagnosticsAttempt.status).toEqual('success'),
+          { interval }
+        );
+        await act(async () => {
+          result.current.openReport(result.current.diagnosticsAttempt.data);
+        });
+        expect(notificationsService.notifyWarning).toHaveBeenCalledTimes(1);
+        expect(notificationsService.getNotifications()).toHaveLength(0);
+
+        jest.clearAllMocks();
+
+        // Wait for the third report to be processed and verify that it does not result in another
+        // notification being created.
+        await waitFor(
+          () => expect(vnet.runDiagnostics).toHaveBeenCalledTimes(3),
+          { interval }
+        );
+        expect(notificationsService.notifyWarning).toHaveBeenCalledTimes(0);
+        expect(notificationsService.getNotifications()).toHaveLength(0);
+      },
+    },
+    {
+      it: "is not shown when the VNet panel is opened and doesn't appear after the panel is closed",
+      controlConnectionsRef: createRef(),
+      mockAppContext: appContext => {
+        jest
+          .spyOn(appContext.vnet, 'runDiagnostics')
+          .mockResolvedValue(
+            new MockedUnaryCall({ report: issuesFoundReport })
+          );
+      },
+      verify: async (
+        { vnet, notificationsService },
+        result,
+        controlConnectionsRef
+      ) => {
+        await waitFor(
+          () =>
+            expect(result.current.diagnosticsAttempt.status).toEqual('success'),
+          { interval }
+        );
+        expect(notificationsService.notifyWarning).not.toHaveBeenCalled();
+
+        // Close the panel and wait for the next run, verify that the notification wasn't sent.
+        await act(async () => controlConnectionsRef.current.close());
+        await waitFor(
+          () => expect(vnet.runDiagnostics).toHaveBeenCalledTimes(2),
+          { interval }
+        );
```

Comment:
This and other similar expectations are technically racy, as there's nothing that guarantees that intervals within `waitFor` are going to be run before intervals from `setInterval` in `vnetContext`. I tried to band-aid this by making `interval` just 10ms and the interval for diag runs to be 50ms in tests. But the event loop can still technically screw this up (I think).

It'd be better to use `jest.useFakeTimers` and manually schedule runs, but I just couldn't get it to work. This is made worse by the fact that you'd typically run `useFakeTimers`/`useRealTimers` from `before`/`after` hooks, which doesn't seem to work well with table tests.